src/api/item.ts

```diff
@@ -74,14 +75,16 @@ export const getAccessibleItems = async (
       .then(({ data }) => data),
   );
 
+type AllOrNothing<T> = T | Partial<Record<keyof T, undefined>>;
```

Comment:
I find this type a bit confusing. This allwos you to have two "linked" propoerties at the top level of the object. Would it be ok to instead next them inside a `geolocation` optional key but have both prop be required ?
```ts
type Test = {
  name: string;
  geolocation?: {
    lat: string;
    lng: string;
  };
}
```
This way if you specify the `geolocation` key you have to provide both values, if you don't then you do not have to provide anything :)